### PR TITLE
Added missing TypedArray functions

### PR DESCRIPTION
**Pull request by JaredWilliams**
TypedArray.of()
TypedArray.from()
TypedArray.prototype.copyWithin()

### CODE REVIEW

**Comment by JaredWilliams** [comment]
I signed it!

**Comment by JaredWilliams** [comment]
I signed it!

**Comment by JaredWilliams** [comment]
I signed it!

File: externs/es6.js [hunk 1]
```diff
@@ -394,6 +394,23 @@ function Int8Array(length, opt_byteOffset, opt_length) {}
 /** @type {number} */
 Int8Array.BYTES_PER_ELEMENT;
 
+/** 
+* @param {Array<number>} source
+* @param {function(this:S, number): number=} opt_mapFn
+* @param {S=} opt_this
+* @template S
+* @return {!Int8Array}
```
Comment (ChadKillingsworth) [inline externs/es6.js]: Shouldn't this be `Int8Array<number>`?

Comment (concavelenz) [inline externs/es6.js]: No, Int8Array isn't generic, it is always IArrayLike<number>

File: externs/es6.js [hunk 2]
```diff
@@ -394,6 +394,23 @@ function Int8Array(length, opt_byteOffset, opt_length) {}
 /** @type {number} */
 Int8Array.BYTES_PER_ELEMENT;
 
+/** 
+* @param {Array<number>} source
```
Comment (MatrixFrog) [inline externs/es6.js]: If there is a good reason to pass null this should be {?Array<number>}. If not then make it {!Array<number>}

Comment (JaredWilliams) [inline externs/es6.js]: It does need to change, just sure to what. 

There are two call scenarios, one with a single argument, which would have to be something iterable returning numbers.

Second is with a source and map function, then the source can by anything iterable as the mapFn provides with numbers. 

Eg. Uint8Array.from(['a', 'aa', 'aaa'], v => v.length)

Comment (MatrixFrog) [inline externs/es6.js]: I admit I haven't looked at this part of the spec but it sounds like this should really be Iterable<number> not Array<number>? In any case I'm fine with submitting this as Array for now and then we can change it to Iterable (or ArrayLike, or whatever it's supposed to really be) later. But, let's make it non-nullable (add the `!`)

**Comment by Dominator008** [comment]
Seems like this will need a rebase.

**Comment by Dominator008** [comment]
Rebased in #1413.

**Pull request closed without merging**